### PR TITLE
[Model Runner V2] Fix -1 placeholder draft token ids for block verification

### PR DESCRIPTION
**Pull request by TheEpicDolphin**
# Summary
The block verification path does not currently guard against -1 placeholder draft token ids, which can be used as padding. This PR simply ensures that these draft tokens are rejected, and that they don't cause OOB for block verification. This was already handled for standard/greedy rejection sampling here: https://github.com/vllm-project/vllm/pull/46533.

# Tests
Added three tests under tests/v1/spec_decode/test_rejection_sampler_utils.py for the -1 placeholder draft tokens case. All pass:
```
python -m pytest tests/v1/spec_decode/test_rejection_sampler_utils.py
Running 36 items in this shard
....................................                                                                                                                                                                                                                                                                            [100%]
36 passed in 10.98s
```

# Benchmarks
### E2E benchmark
Model Runner V2, GB200 tp=1, SPEED-Bench `throughput_2k`, 128 prompts × 1024 output
tokens, concurrency 32, fixed seed, `--kernel-config.enable_flashinfer_autotune=False`.
`before` = `cb8104839c` (parent commit); only `rejection_sampler_utils.py` is swapped.

| model | rejection + draft sample method | accept len after | accept len before | Δ | accept rate % after | accept rate % before | Δ (pp) | out tok/s Δ | median ITL Δ |
|---|---|---|---|---|---|---|---|---|---|
| Llama-3-8B + EAGLE (K=4) | `standard` + `greedy` | 2.170 | 2.180 | -0.010 | 29.33 | 29.44 | -0.11 | -3.28% | +0.13% |
| Llama-3-8B + EAGLE (K=4) | `standard` + `probabilistic` | 2.230 | 2.220 | +0.010 | 30.77 | 30.53 | +0.24 | +0.44% | +0.00% |
| Llama-3-8B + EAGLE (K=4) | `block` + `greedy` | 2.190 | 2.210 | -0.020 | 29.76 | 30.15 | -0.39 | -2.22% | +0.13% |
| Llama-3-8B + EAGLE (K=4) | `block` + `probabilistic` | 2.210 | 2.210 | +0.000 | 30.34 | 30.28 | +0.06 | +0.20% | +0.13% |
| Qwen3-8B + DFlash-b16 (K=7) | `standard` + `greedy` | 2.200 | 2.210 | -0.010 | 17.19 | 17.36 | -0.17 | +0.05% | -0.24% |
| Qwen3-8B + DFlash-b16 (K=7) | `standard` + `probabilistic` | 2.160 | 2.160 | +0.000 | 16.62 | 16.63 | -0.01 | +0.90% | +0.35% |
| Qwen3-8B + DFlash-b16 (K=7) | `block` + `greedy` | 2.220 | 2.220 | +0.000 | 17.40 | 17.44 | -0.04 | -0.31% | -0.24% |
| Qwen3-8B + DFlash-b16 (K=7) | `block` + `probabilistic` | 2.190 | 2.198 | -0.008 | 17.01 | 17.12 | -0.11 | -0.48% | +0.09% |

Acceptance is at parity: max |Δ acceptance length| is 0.020, signs mixed, no systematic
direction. `block`+`probabilistic` on DFlash is reported as the mean of 4 paired runs
(per-run Δ: +0.040, -0.010, -0.030, -0.030; sd 0.033). Median ITL — where per-step
sampling cost lands — is within 0.35% everywhere.

### Kernel microbenchmark (V=129280, K=8, batch 256, block verification, full draft logits)
| | before | after | Δ |
|---|---|---|---|
| no placeholders | 0.901 ms | 0.909 ms | +0.9% |
| half placeholders | 0.894 ms | 0.714 ms | **-20%** |
| all placeholders | 0.845 ms | 0.539 ms | **-36%** |

### CODE REVIEW

**Review by njhill — COMMENTED** [review]
Thanks @TheEpicDolphin, good catch!

Claude suggested a bit of related further rework, which also fixes a subtle bug, ptal: https://github.com/njhill/vllm/commit/88a1769f25729731cf3898079bf3475ab8135d37

**Comment by TheEpicDolphin** [comment]
@njhill Thanks for the catch! I folded in your suggestions, and added some unit tests to `test_rejection_sampler_utils.py`  covering the -1 placeholder handling.

I refactored things a bit so that the rejection sampler loop skips verifying the remaining steps once the first -1 placeholder token is encountered. That way, we don't waste compute on those steps. I also had to make `_resample_kernel` handle the case where the rejected draft was a -1 placeholder, and use the target logits instead of the `max(p(x) - q(x), 0)` residual, which is incorrect because the -1 placeholder won't have probability mass to subtract.

**Review by njhill — APPROVED** [review]
Thanks @TheEpicDolphin the updates look great
